src/NUnitFramework/framework/Constraints/SubstringConstraint.cs

```diff
@@ -56,13 +53,12 @@ protected override bool Matches(string actual)
         }
 
         /// <summary>
-        /// Flag the constraint to use the supplied StringComparison type.
+        /// Overrides the <see cref="StringConstraint.IgnoreCase"/> setting and instead uses the specified comparison.
         /// </summary>
-        /// <param name="comparisonType ">The StringComparison type to be used.</param>
-        /// <returns>Self.</returns>
-        public SubstringConstraint Using(StringComparison comparisonType)
+        public SubstringConstraint Using(StringComparison? comparisonType)
         {
-            this.comparisonType = comparisonType;
+            this.comparisonType = comparisonType ?? (caseInsensitive ? 
+                StringComparison.CurrentCultureIgnoreCase : StringComparison.Ordinal);
```

Comment:
@CharliePoole This is necessary for back-compat. If we want to change this, we need a design discussion.

Currently, `Assert.That("ss", Contains.Substring("ß"))` fails and `Assert.That("ss", Contains.Substring("ß").IgnoreCase)` succeeds. If we made both `CurrentCulture*`, they would both succeed. If we made both `Ordinal*`, the would both fail.